mobile/lib/ui/collections/flex_grid_view.dart

```diff
@@ -23,49 +39,255 @@ class CollectionsFlexiGridViewWidget extends StatelessWidget {
   final bool shrinkWrap;
   final String tag;
 
+  final AlbumViewType albumViewType;
+  final bool enableSelectionMode;
+  final bool shouldShowCreateAlbum;
+  final SelectedAlbums? selectedAlbums;
+  final double scrollBottomSafeArea;
+
   const CollectionsFlexiGridViewWidget(
     this.collections, {
-    this.displayLimitCount = 10,
+    this.displayLimitCount = 9,
     this.shrinkWrap = false,
     this.tag = "",
+    this.enableSelectionMode = false,
     super.key,
+    this.albumViewType = AlbumViewType.grid,
+    this.shouldShowCreateAlbum = false,
+    this.selectedAlbums,
+    this.scrollBottomSafeArea = 8,
   });
 
+  @override
+  State<CollectionsFlexiGridViewWidget> createState() =>
+      _CollectionsFlexiGridViewWidgetState();
+}
+
+class _CollectionsFlexiGridViewWidgetState
+    extends State<CollectionsFlexiGridViewWidget> {
+  bool isAnyAlbumSelected = false;
+  late StreamSubscription<ClearAlbumSelectionsEvent>
+      _clearAlbumSelectionSubscription;
+
+  @override
+  void initState() {
+    _clearAlbumSelectionSubscription =
+        Bus.instance.on<ClearAlbumSelectionsEvent>().listen((event) {
+      if (mounted) {
+        setState(() {
+          isAnyAlbumSelected = false;
+        });
+      }
+    });
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    _clearAlbumSelectionSubscription.cancel();
+    super.dispose();
+  }
+
+  Future<void> _toggleAlbumSelection(Collection c) async {
+    await HapticFeedback.lightImpact();
+    widget.selectedAlbums!.toggleSelection(c);
+    setState(() {
+      isAnyAlbumSelected = widget.selectedAlbums!.albums.isNotEmpty;
+    });
+  }
+
+  Future<void> _navigateToCollectionPage(Collection c) async {
+    final thumbnail = await CollectionsService.instance.getCover(c);
+    // ignore: unawaited_futures
+    routeToPage(
+      context,
+      CollectionPage(
+        CollectionWithThumbnail(c, thumbnail),
+      ),
+    );
+  }
+
   @override
   Widget build(BuildContext context) {
+    return widget.albumViewType == AlbumViewType.grid
+        ? _buildGridView(context, const ValueKey("grid_view"))
+        : _buildListView(context, const ValueKey("list_view"));
+  }
+
+  Widget _buildGridView(BuildContext context, Key key) {
     final double screenWidth = MediaQuery.of(context).size.width;
```

Comment:
Better to use more specific alternative `MediaQuery.sizeOf(context)`. See why [here](https://stackoverflow.com/questions/76474858/with-flutter-3-10-which-is-the-better-way-to-get-the-size-with-mediaquery).

FYI it doesn't make any performance differences here from a quick glance.